pkg/cmd/historical_logs.go

```diff
@@ -107,40 +118,317 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
+	if len(o.Tail) > 0 {
+		tail, err := strconv.Atoi(o.Tail[0 : len(o.Tail)-1]) //extract numeric value. For example, extract 50 from 50s or 10 from 10m
+		if err != nil {
+			return fmt.Errorf("an invalid \"tail\" value was entered: %v", err)
+		}
 
-	if err != nil {
-		return fmt.Errorf("Request Failed: %v", err)
+		timeUnit := o.Tail[len(o.Tail)-1] //Last character (time unit) is 's'(seconds),'m'(minutes),'h'(hours),'d'(days)
+		endTime := time.Now().UTC()
+		var startTime time.Time
+
+		switch timeUnit {
+		case 's':
+			startTime = endTime.Add(time.Duration(-tail) * time.Second).UTC()
+		case 'm':
+			startTime = endTime.Add(time.Duration(-tail) * time.Minute).UTC()
+		case 'h':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour).UTC()
+		case 'd':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour * 24)
+		default:
+			return fmt.Errorf("invalid time unit entered in \"tail\". please enter s, m, h, or d as time unit")
+		}
+
+		o.StartTime = startTime.UTC().Format(time.RFC3339Nano)
+		o.EndTime = endTime.UTC().Format(time.RFC3339Nano)
 	}
 
-	client := &http.Client{}
-	res, err := client.Do(req)
-	if err != nil {
-		return fmt.Errorf("Response Error: %v", err)
+	var podList []string
+
+	if len(o.Deployment) > 0 {
+		err = GetDeploymentPodsList(clientset, o.Deployment, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(o.DaemonSet) > 0 {
+		err = GetDaemonSetPodsList(clientset, o.DaemonSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
 	}
 
-	responseBody, err := ioutil.ReadAll(res.Body)
+	if len(o.StatefulSet) > 0 {
+		err = GetStatefulSetPodsList(clientset, o.StatefulSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	var logList []string
+	if len(podList) > 0 {
+		for _, pod := range podList {
+
+			o.Podname = pod
+			response,err := o.makeHttpRequest(ApiUrl)
+
+			if err!=nil{
+				return err
+			}
+
+			responseBody, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response: %v", err)
+			}
+
+			err = getLogs(responseBody, &logList)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		response,err := o.makeHttpRequest(ApiUrl)
+		if err!=nil{
+			return err
+		}
+		responseBody, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response: %v", err)
+		}
+
+		err = getLogs(responseBody, &logList)
+		if err != nil {
+			return err
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *LogParameters) makeHttpRequest(baseUrl string) (*http.Response,error) {
+
+	var urlBuilder strings.Builder
+	urlBuilder.WriteString(baseUrl)
+	numParameters := 0
+	if len(o.Podname) >0{
+		urlBuilder.WriteString("?")
+		urlBuilder.WriteString("podname="+o.Podname)
+		numParameters = numParameters + 1
```

Comment:
@AkshayBhansali18 just curious why do we need to have count of parameters?

Reply:
The main purpose of `numParameters` is to identify the first parameter that goes into the URL. I thought it might be useful to also make a note of the count of parameters. But yes, it could be replaced with a `bool` flag.